gegelatilib/include/importer/tpgGraphDotImporter.h

```diff
@@ -0,0 +1,400 @@
+#ifndef TPG_GRAPH_DOT_IMPORTER_H
+#define TPG_GRAPH_DOT_IMPORTER_H
+
+#include <stdexcept>
+#include <map>
+#include <string>
+#include <memory>
+#include <inttypes.h>
+#include <regex>
+#include <fstream>
+#include <cstdio>
+
+#include "learn/learningEnvironment.h"
+#include "tpg/tpgVertex.h"
+#include "tpg/tpgTeam.h"
+#include "tpg/tpgAction.h"
+#include "tpg/tpgEdge.h"
+#include "tpg/tpgGraph.h"
+
+/**
+*	\brief Maximum number of characters that can be read in a single line.
+*/
+#define MAX_READ_SIZE 1024
+
+namespace Importer {
+	/**
+	* \brief Class used to import a TPG graph from a dot file.
+	* It should be able to import a whole Learning agent object.
+	*/
+	class TPGGraphDotImporter {
+	protected:
+		/**
+		* \brief File in which the dot content is read during import.
+		*/
+		std::ifstream pFile;
+
+		/**
+		* \brief last Line read from file 
+		*
+		*	helps to pass a line that didn't match any regex to another function
+		*/
+		std::string lastLine;;
+
+		/**
+		* \brief The environment in which the TPGGRAPH will be built
+		*/
+		Environment env;
+
+		/**
+		* \brief TPGGraph imported from dot file.
+		*/
+		TPG::TPGGraph tpg;
+
+		/**
+		* \brief Map associating pointers to TPGVertex to an integer ID.
+		*
+		* This map is used to associate an unique id to a tpg vertex and to 
+		* keep track of the pointers while restoring the TPGGraph described in a dot file
+		*/
+		std::map<uint64_t, const TPG::TPGVertex*> vertexID;
+
+		/**
+		* \brief Map associating pointers to Program to an integer ID.
+		*
+		* This map is used to associate an unique id to a tpg program and to 
+		* keep track of the pointers while restoring the TPGGraph described in a dot file
+		*/
+		std::map<uint64_t, std::shared_ptr<Program::Program>> programID;
+		
+		/**
+		* \brief Map associating pointers to TPGVertex representing actions
+		* to the corresponding action 
+		*
+		* This map is used to ensure that identical actions are not created
+		* moore than once.
+		*/
+		std::map<uint64_t, const TPG::TPGVertex*> actionID;
+
+		/**
+		* \brief Map associating actions to the corresponding action ID
+		*
+		* This map is here is used to access the correct TPGVertex while linking an action.
+		*/
+		std::map<uint64_t, uint64_t> actionLabel;
+
+		/**
+		*	\brief string used to spot the end of a line in the program description.
+
+		*/
+		const std::string lineSeparator;
+
+		/**
+		*	\brief contains the regex to identify a team declaration
```

Comment:
Upper case at the beginning of sentence.
```suggestion
		*	\brief Contains the regex to identify a team declaration
```